### PR TITLE
Let the mmt4d ukernel declare its code paths so we know at compile time when it's going to be fast.

### PR DESCRIPTION
**Pull request by bjacob**

### CODE REVIEW

File: compiler/plugins/target/LLVMCPU/LLVMCPUTarget.cpp [hunk 1]
```diff
@@ -515,6 +517,24 @@ class LLVMCPUTargetBackend final : public TargetBackend {
                         variantOp.getName(), ".codegen.bc", *llvmModule);
     }
 
+    {
+      const char* codegenModuleFileEnv = getenv("CODEGEN_MODULE_FILE");
```
Comment (benvanik) [inline compiler/plugins/target/LLVMCPU/LLVMCPUTarget.cpp]: what exactly is this for? we really don't want to be adding env vars if we can avoid it

File: compiler/plugins/target/LLVMCPU/LLVMCPUTarget.cpp [hunk 2]
```diff
@@ -515,6 +517,24 @@ class LLVMCPUTargetBackend final : public TargetBackend {
...
+      if (codegenModuleFileEnv) {
+        auto codegenModuleBuffer = llvm::MemoryBuffer::getFile(codegenModuleFileEnv);
+        if (!codegenModuleBuffer) {
+          fprintf(stderr, "FATAL: failed to open %s\n", codegenModuleFileEnv);
```
Comment (benvanik) [inline compiler/plugins/target/LLVMCPU/LLVMCPUTarget.cpp]: we also don't want to be doing this in the compiler - I think this all should probably be deleted unless converted to something better built to be in a compiler

Comment (bjacob) [reply compiler/plugins/target/LLVMCPU/LLVMCPUTarget.cpp]: Accidentally committed, removed in latest force push

**Comment by bjacob** [comment]
*la la la @benvanik is very busy with important GPU stuff ,  not looking here la la la*

**Pull request closed without merging**